geoh5py/shared/__init__.py

```diff
@@ -25,3 +26,5 @@
 
 INTEGER_NDV = -2147483648
 FLOAT_NDV = 1.17549435e-38
+
+logger = logging.getLogger(__name__)
```

Comment:
**suggestion:** create logger in each module

This will create a logger for this specific module "shared", used everywhere. this is not desirable.
One benefit of using logger is that it accept a global configuration where the behaviour can be tweaked for each module.

A client application can configure the logging, and turn off log from some modules, or change the logging level per module, or redirect, reformat, etc...

From [advanced-logging-tutorial](https://docs.python.org/3/howto/logging.html#advanced-logging-tutorial):

> A good convention to use when naming loggers is to use a module-level logger, in each module which uses logging, named as follows:
> ```python
> logger = logging.getLogger(__name__)
> ```
> This means that logger names track the package/module hierarchy, and it’s intuitively obvious where events are logged just from the logger name.

Reply:
Should you want to have a common logger for all the 'object'?
Or a logger for every specific module using 'print'?

Reply:
the recommendation is to have a logger per module.
So each file will call its own: `logger = logging.getLogger(__name__)`

It is safe to call it multiple times in a file: it returns the same logger instance.
It can also be placed once the top of each file that needs it, after imports